### ISSUE TITLE
Metadata gives wrong error if Redis is restarted after REST API call

### ISSUE THREAD

**Issue by dmocek** [edgexfoundry/edgex-go#4463]
# 🐞 Bug Report

### Affected Services [**REQUIRED**]

 The issue is located in: Metadata

### Is this a regression?

 I don't believe so.

### Description and Minimal Reproduction [**REQUIRED**]
Steps:
1. Install and run EdgeX Jakarta.
2. Create at least one Device Profile.
3. Run this REST API: 'curl -s -X GET http://localhost:59881/api/v2/deviceprofile/all'
4. Stop Redis using 'docker stop edgex-redis'.
5. Run the same command from step 3.
6. Recieve 'HTTP request timeout'.

These steps return the expected error:
1. Install and run EdgeX Jakarta.
2. Create at least one Device Profile.
3. Stop Redis using 'docker stop edgex-redis'.
4. Run this REST API: 'curl -s -X GET http://localhost:59881/api/v2/deviceprofile/all'
6. Recieve '{"apiVersion":"v2","message":"failed to get member number with command ZCARD from md|dp","statusCode":500}'

It appears that making a REST API call before stopping Redis causes an incorrect error.

## 🔥 Exception or Error
<pre><code>

'HTTP request timeout'
</code></pre>

## 🌍  Your Environment
**Deployment Environment:**
Oracle Linux 9
**EdgeX Version [**REQUIRED**]:**
v2.1.1

**Anything else relevant?**

**Comment by hahattan** [comment]
Hi @dmocek are you seeing this error consistently because I cannot reproduce it (HTTP request timeout) in my environment.
From the reproduction steps are you running step4 before step3 finished ?

**Comment by cloudxxx8** [comment]
@dmocek we are going to release v3 very soon.  please help us resolve this bug before the code freeze date (May 10th), or this can only be posponed.

**Comment by aviralx10** [comment]
Hi @cloudxxx8 @chr1shung . I am facing the same issue as @dmocek . 
To answer your question @chr1shung : yes, i made sure that the step 3 is finished properly before moving on to step 4.

**Comment by dmocek** [comment]
Are you using EdgeX Jakarta?  Is it possible you're using a newer version of EdgeX where the issue is fixed?

**Comment by dmocek** [comment]
Any update on this?

**Comment by cloudxxx8** [comment]
@dmocek Chris left this project.  We would assign someone else to look into this issue.

**Comment by lenny-intel** [comment]
@dmocek , I just tested this scenario using Jakarta 2.1.1 and after a few seconds Core Metadata was responding correctly.
I followed your steps above using https://github.com/edgexfoundry/edgex-compose/blob/v2.1.1/docker-compose-no-secty.yml 

One difference I get is on step 6 in first section, I get the `failed to get member number with command ZCARD from md|dp","statusCode":500` error, not a timeout error.

Once I restart Redis, I may get strange results (same error above, empty list) if request too quickly, but after a few seconds I get the expected results.
